### PR TITLE
ISSUE #3 - Allow busterbunny to emit statistics at specified interval

### PR DESCRIPTION
**Pull request by johndietz**
Added config.stats which will emit busterbunny statistics at specified number of seconds if provided
Unit testing added
https://github.com/GannettDigital/busterbunny.js/issues/3

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -62,7 +62,8 @@ var config = {
             }
         ],
         exchange: 'i.write.2.this1'
-    }
+    },
+    stats: 10
```
Comment (binaryalchemist) [inline README.md]: Let's call this `statsInterval`.

File: src/busterbunny.js [hunk 2]
```diff
@@ -49,6 +50,15 @@ module.exports = (function() {
             EVENT_NACKED: 'event-nacked'
         });
 
+        if(config.stats){
+            function emitStats()
+            {
```
Comment (binaryalchemist) [inline src/busterbunny.js]: This curly should be on line 37 ... cuz JavaScript.

File: src/busterbunny.js [hunk 3]
```diff
@@ -49,6 +50,15 @@ module.exports = (function() {
...
+                self.emit(self.EVENTS.STATS, _stats);
```
Comment (binaryalchemist) [inline src/busterbunny.js]: Check for listeners of self.EVENTS.STATS before emitting.

Comment (enigmatic00) [inline src/busterbunny.js]: Also, should a copy of the `_stats` be emitted instead ? The `_stats_` object may mutate in the time that it is emitted & consumed. 

@kakashif & I ran into this with another library as well where an emitted array mutated while we were processing it

Comment (binaryalchemist) [inline src/busterbunny.js]: @kakashif is right ... this should be a copy of the object.

Comment (kakashif) [inline src/busterbunny.js]: Correct, if you care to use the _stats object beyond this function, you should create at the very minimum a shallow copy before emitting.  If you care about data found for individual items in the array, then you would need to do a deep copy.

**Comment by binaryalchemist** [comment]
Looks good.  This would close #1.

**Pull request merged**